lighthouse-core/report/html/renderer/i18n.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * @license
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+/* globals self, URL */
+
+// Not named `NBSP` because that creates a duplicate identifier (util.js). 
+const NBSP2 = '\xa0';
+
+class I18n {
+  /**
+   * @param {LH.Locale} locale
+   */
+  constructor(locale) {
+    this.setNumberDateLocale(locale);
+    this._numberDateLocale = locale;
+    this._numberFormatter = new Intl.NumberFormat(locale);
+  }
+
+  /**
+   * @param {LH.I18NRendererStrings} strings
+   */
+  setStrings(strings) {
```

Comment:
feels a little weird to pair this with a getter property, I would just expect `set strings()` or `getStrings()`